src/bundles/experiments.js

```diff
@@ -0,0 +1,97 @@
+const ACTIONS = {
+  EXP_TOGGLE: 'EXPERIMENTS_EXP_TOGGLE'
+}
+
+// setup experiments
+const EXPERIMENTS = {
+  npm: {
+    action: async () => null,
+    enabled: false,
+    actionUrls: [
+      {
+        url: 'https://github.com/ipfs-shipyard/npm-on-ipfs',
+        key: 'readMoreUrl'
+      },
+      {
+        url: 'https://github.com/ipfs-shipyard/npm-on-ipfs/issues',
+        key: 'issueUrl'
+      },
+      {
+        url: 'https://github.com/ipfs-shipyard/npm-on-ipfs',
+        key: 'feedbackUrl'
+      }
+    ],
+    desktop: false
+  },
+  tpd: {
+    action: async () => null,
+    enabled: false
+  },
+  tpda: {
+    action: async () => null,
+    enabled: false
+  },
+  tpdb: {
+    action: async () => null,
+    enabled: false
+  }
+}
+
+// helpers
+const createAction = (type, key) => ({
+  type: type,
+  payload: { key }
+})
+
+const objAsArr = obj =>
+  Object.keys(obj).map(i => ({
+    ...obj[i],
+    key: i
+  }))
+
+const isEnabled = (state, key) => {
+  if (state && state[key]) {
+    return state[key].enabled
+  }
+}
+
+const toggleEnabled = (state, key) => {
+  return {
+    ...state,
+    [key]: {
+      ...state[key],
+      enabled: !isEnabled(state, key)
+    }
+  }
+}
+//
+
+export default {
+  name: 'experiments',
+  // persist all actions
+  persistActions: Object.values(ACTIONS),
+  reducer: (state = {}, action) => {
+    if (action.type === ACTIONS.EXP_TOGGLE) {
+      return toggleEnabled(state, action.payload.key)
+    }
+    return state
+  },
+  doToggleAction: (key, enabled) => async ({ dispatch }) => {
+    if (!key) return
+
+    const experiment = EXPERIMENTS[key]
+    const dispatchAction = createAction(ACTIONS.EXP_TOGGLE, key)
+
+    if (typeof experiment.action === 'function') {
+      await Promise.resolve(experiment.action(enabled))
+        .then(res => dispatch(dispatchAction))
+        .catch(err => console.error(err))
```

Comment:
note: if there is a better way to inform the user about errors via a notification or modal we should do so here.

Reply:
You can use the `notify` bundle to trigger notifications.

Check out [this PR](https://github.com/ipfs-shipyard/ipfs-webui/pull/969) from @hacdias to see how to use it.

Reply:
Thanks, I'll take a look!

Reply:
Is there already a dispatch action I can use to display display a toast notification? It's not clear from the example.

Reply:
the pattern at the moment is to centrally manage all notifications from the notifications bundle, like https://github.com/ipfs-shipyard/ipfs-webui/pull/969/files#diff-7f24a40e51bf52d47e7cf3f255a6a6a8

I don't love that pattern as it's a bit harder to reason about, so we may change it, but that's where they go today.

Reply:
I feel it's ok to release this without notifications and revisit in another PR later on

Reply:
👍